apps/web/src/utils/redirectIfNotNameOwner.ts

```diff
@@ -0,0 +1,18 @@
+import { BaseName } from '@coinbase/onchainkit/identity';
+import {
+  getBasenameAddress,
+  getBasenameEditor,
+  getBasenameOwner,
+} from 'apps/web/src/utils/usernames';
+import { redirect } from 'next/navigation';
+
+export async function redirectIfNotNameOwner(username: BaseName) {
+  const address = await getBasenameAddress(username);
+  const editor = await getBasenameEditor(username);
+  const owner = await getBasenameOwner(username);
```

Comment:
```suggestion
  const [address, editor, owner] = await Promise.all([getBasenameAddress(username), getBasenameEditor(username), getBasenameOwner(username)];
```

Even better would be to early exit after any of these return falsey.

Reply:
good call